src/libserver/milter.c

```diff
@@ -1715,10 +1722,19 @@ rspamd_milter_extract_single_header (struct rspamd_milter_session *session,
 			else {
 				/* Calculate negative offset */
 
-				if (-idx <= priv->cur_hdr) {
+				if (idx == -1) {
+					rspamd_milter_send_action (session,
+							RSPAMD_MILTER_ADDHEADER,
+							hname, hvalue);
+				}
+				else if (-idx <= priv->cur_hdr) {
+					/*
+					 * Note: We should account MTA's own "Received:" field
+					 * which wasn't passed by Milter's header command.
+					 */
```

Comment:
👍
However, some MTA can include that header in Milter dialog sadly. Postfix and Sendmail are fine here to my very best knowledge.

Reply:
I also guessed it possible (I tested only with Postfix), so I isolated the `-1` case.  Thanks for confirming my concern.